gridmet.py

```diff
@@ -0,0 +1,152 @@
+# Libraries
+import pandas as pd
+import pickle
+import time
+import hvplot.pandas
+import hvplot.xarray
+import geopandas as gpd
+import grd2shp_xagg
+import xagg as xa
+import xarray as xr
+
+# Variables
+gdf_nhru02_path = './data/nhru_02/nhru_02.shp'
+start_date = '1979-01-01'
+end_date = '1989-01-01'
+lon_min = -79
+lon_max = -67
+lat_min = 36
+lat_max = 42
+output_path = './data/'
+## official list of variables needed for drb-inland-salinity model
+data_vars_shrt_all = ['tmmx', 'tmmn', 'pr', 'srad', 'vs','rmax','rmin','sph']
+data_vars_long_all = ['daily_maximum_temperature', 'daily_minimum_temperature',
+         'precipitation_amount', 'daily_mean_shortwave_radiation_at_surface',
+         'daily_mean_wind_speed', 'daily_maximum_relative_humidity',
+         'daily_minimum_relative_humidity','daily_mean_specific_humidity']
+
+## Shorter list of data variables for testing:
+data_vars_grd2shp_shrt = ['tmmn', 'vs']
+
+## Full function
+def g2shp_regridding(polygon_file_path,
+                     output_data_folder,
+                     var_short,
+                     start_date, end_date,
+                     lon_min, lon_max, lat_min, lat_max,
+                     g2s_file_prefix = 'tmp'):
+    """
+    :param str polygon_file_path: path to hru shapefile used for regridding of gridMET data
+    :param str output_data_folder: folder location where outputs such as weights pickle file and netcdf output file will reside
+    :param str/list var_short: data variable short name or list of data variable short names. Must be one of the following: ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    :param str start_date: Start date of data collection yyyy-mm-dd
+    :param str end_date: End date of data collection yyyy-mm-dd
+    :param str lon_min: bbox of aoi longitude min
+    :param str lon_max: bbox of aoi longitude max
+    :param str lat_min: bbox of aoi latitude min
+    :param str lat_max: bbox of aoi latitude max
+    :param str g2s_file_prefix: prefix used to save the grd2shp_xarray
+    :return: g2s array
+    """
+
+    ## Create dictionary of short variable names with official long variable name
+    data_vars_shrt_all = ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    data_vars_long_all = ['daily_maximum_temperature', 'daily_minimum_temperature',
+                          'precipitation_amount', 'daily_mean_shortwave_radiation_at_surface',
+                          'daily_mean_wind_speed', 'daily_maximum_relative_humidity',
+                          'daily_minimum_relative_humidity', 'daily_mean_specific_humidity']
+
```

Comment:
To me it makes more sense to use the global variables above and then you can just define them once (lines 22-26) and these can be deleted.